vmas/scenarios/navigation.py

```diff
@@ -283,6 +290,98 @@ def extra_render(self, env_index: int = 0) -> "List[Geom]":
         return geoms
 
 
+class HeuristicPolicy(BaseHeuristicPolicy):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.clf_epsilon = 0.2  # Exponential CLF convergence rate
+        self.cbf_eta = 1  # CBF class-K parameter
+        self.cbf_zeta = 1  # CBF class-K parameter
+        self.clf_slack = 100.0  # weights on CLF-QP slack variable
+
+    def compute_action(self, observation: Tensor, u_range: Tensor) -> Tensor:
+        """
+        QP inputs:
+        These values need to computed apriri based on observation before passing into QP
+
+        V: Lyapunov function value
+        lfV: Lie derivative of Lyapunov function
+        lgV: Lie derivative of Lyapunov function
+        CLF_slack: CLF constraint slack variable
+
+        QP outputs:
+        u: action
+        CLF_slack: CLF constraint slack variable, 0 if CLF constraint is satisfied
+        """
+        self.n_env = observation.shape[0]
+        self.device = observation.device
+        agent_pos = observation[:, :2]
+        agent_vel = observation[:, 2:4]
+        goal_pos = (-1.0) * (observation[:, 4:6] - agent_pos)
+
+        # Pre-compute tensors for the CLF and CBF constraints,
+        # Lyapunov Function from: https://arxiv.org/pdf/1903.03692.pdf
+
+        # Laypunov function
+        V_value = (
+            (agent_pos[:, 0] - goal_pos[:, 0]) ** 2
```

Comment:
For the X and Y dimensions (0 and 1), you can use `vmas.simulator.utils.X` and the same for Y to increase readability

Reply:
Cool, I will modify them.